### PR TITLE
feat: APPS-3058   Create FTVA variation of SectionStaffArticleList

### PR DESCRIPTION
**Pull request by jendiamond**
Connected to [APPS-3058](https://jira.library.ucla.edu/browse/APPS-3058)
This FTVA styled component will be used on the [EventSeries Listing page](https://www.figma.com/design/L8e1czKo8a8px368ayyPZ4/%5BFTVA%5D-Final-Design-with-Annotations?node-id=11-1445&node-type=canvas&t=NVJ5v7jVQpNMxCsh-0).

https://deploy-preview-653--ucla-library-storybook.netlify.app/?path=/story/section-staff-article-list--ftva-current-entries-with-dates

**Notes:**
+ `src/lib-components/BlockStaffArticleList.vue`
    + Moved all utils together
    + Add props
        + startDate
        + endDate
        + ongoing
    + Add the Theme and classes for Default & FTVA scss
    + Add the logic for the dates
        + parsedDateDisplay
    + Update the date formatting in `formatSeriesDates`
+ `src/lib-components/SectionStaffArticleList.vue`
    + Add calls to
        + startDate
        + endDate
        + ongoing
    + Add the Theme and classes for Default & FTVA scss
+ `src/stories/BlockStaffArticleList.stories.js`
    + Add stories and data for 
        + FtvaDateRange
        + FtvaSameStartEndDate
        + FtvaOngoing
+ `src/stories/SectionStaffArticleList.stories.js`
    + Add stories and data for 
        + FtvaDateRange
        + FtvaSameStartEndDate
        + FtvaOngoing
+ `src/styles/_base.scss`
    + import `"default/_block-staff-article-item";`
    + import `"default/_section-staff-article-list";`
+ `src/styles/_ftva-theme.scss`
    + import `"ftva/_block-staff-article-item";`
    + import `"ftva/_section-staff-article-list";`
+ Update styles
    + `src/styles/default/_block-staff-article-item.scss`
    + `src/styles/default/_section-staff-article-list.scss`
    + `src/styles/ftva/_block-staff-article-item.scss`
    + `src/styles/ftva/_section-staff-article-list.scss`
+ `src/types/types.ts`
    + Update the BlockStaffArticleListItemType with
        + startDate?: string
        + endDate?: string
        + ongoing?: boolean
+ `src/utils/formatEventSeriesDates.js`
    + Updates the Range to have the full year on both sides
Screenshot 2024-11-22 at 11 19 05 AM

**Checklist:**
- [x] I checked that it is working locally in the storybook
- [ ] I checked that it is working locally in the 
library-website-nuxt dev server
    + My NPM Global is messed up so I can't check this.
- [x] I added a screenshot of it working
- [x] UX has reviewed and approved this
- [x] I used a conventional commit message
- [x] I assigned myself to this PR
Screenshot 2024-11-20 at 1 40 22 PM

Screenshot 2024-11-20 at 1 40 41 PM

[APPS-3058]: https://uclalibrary.atlassian.net/browse/APPS-3058?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/lib-components/BlockStaffArticleList.vue [hunk 1]
```diff
@@ -76,18 +98,48 @@ const parsedTextAll = computed(() => {
...
+  else if (props.endDate)
+    return formatSeriesDates(props.startDate, props.endDate, 'shortWithYear')
+  else if (props.startDate)
+    return formatSeriesDates(props.startDate, props.startDate)
+  else
+    return null
+})
 </script>
 
+<!-- header-smart
```
Comment (pghorpade) [inline src/lib-components/BlockStaffArticleList.vue:115]: Remove this commented code added.

File: src/lib-components/BlockStaffArticleList.vue [hunk 2]
```diff
@@ -55,6 +59,24 @@ const props = defineProps({
     type: Number,
     default: 0,
   },
+  startDate: {
+    type: String,
+    required: false,
```
Comment (pghorpade) [inline src/lib-components/BlockStaffArticleList.vue:64]: don't need required as false here as it isthe  default behavior of props

File: src/lib-components/BlockStaffArticleList.vue [hunk 3]
```diff
@@ -55,6 +59,24 @@ const props = defineProps({
...
+  },
+  endDate: {
+    type: String,
+    required: false,
```
Comment (pghorpade) [inline src/lib-components/BlockStaffArticleList.vue:68]: don't need required as false here as it isthe  default behavior of props

File: src/lib-components/BlockStaffArticleList.vue [hunk 4]
```diff
@@ -76,18 +98,48 @@ const parsedTextAll = computed(() => {
     ? removeHtmlTruncate(props.description, 250)
     : ''
 })
+
+// Display date based on which data is provided
+const parsedDateDisplay = computed(() => {
+  if (props.ongoing === true)
```
Comment (pghorpade) [inline src/lib-components/BlockStaffArticleList.vue:104]: if (props.ongoing)

File: src/lib-components/BlockStaffArticleList.vue [hunk 5]
```diff
@@ -76,18 +98,48 @@ const parsedTextAll = computed(() => {
...
+    return 'Ongoing'
+  else if (props.endDate)
```
Comment (pghorpade) [inline src/lib-components/BlockStaffArticleList.vue:106]: this is not clear to me
 else if (props.endDate)
    return formatSeriesDates(props.startDate, props.endDate, 'shortWithYear')
  else if (props.startDate)
    return formatSeriesDates(props.startDate, props.startDate)
    
    why are we checking in separately if condition startdate and endate exists and formatting it differently?

Comment (pghorpade) [reply src/lib-components/BlockStaffArticleList.vue:106]: also should this date display be specific for ftva theme and ftvaeventseries section handle

**Review by pghorpade — CHANGES_REQUESTED** [review]

File: src/lib-components/BlockStaffArticleList.vue (hunk 5 above)
Comment (jendiamond) [reply src/lib-components/BlockStaffArticleList.vue:103]: I wasn't sure if we wanted to be that specific in the component

**Review by pghorpade — APPROVED** [review]
👍

**Pull request merged**